### ISSUE TITLE
Media conversion stuck on state "coverting"

### ISSUE THREAD

**Issue by dominox71** [cwilby/nextcloud-workflow-media-converter#290]
## Steps to reproduce
1. Install nextcloud on Debian 11
2. Install Automated media conversion
3. Chose folder with file to convert
4. Converting stuck on state "converting"

### Expected behaviour
Convert from .mkv to .mp4

### Actual behaviour
Stuck on state "converting" no matter which user I use and how long I'm waiting, after 24h file is still converting. But when I use app "Video converter" everything works fine and file is converted immediately (test Sample.mkv which i want to convert has only 4mb).

## Workflow Media Converter app

**Workflow Media Converter app version:** (see apps admin page: `/index.php/settings/apps`)
1.7.0
## Server configuration

**Operating system**: Ubuntu/RedHat/...
Debian 11
**Web server:** Apache/Nginx
Apache
**Database:** MySQL/Maria/SQLite/PostgreSQL
mysql 10.5.18
**PHP version:**
7.4.33
**Nextcloud Version:** (see admin page)
Nextcloud Hub 3 (25.0.2)

```
Login as admin user into your Nextcloud and access
http://example.com/index.php/settings/integrity/failed
paste the results here.
```
No errors have been found.
**List of activated apps:**
Enabled:
  - activity: 2.17.0
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - external: 5.0.0
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_external: 1.17.0
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - issuetemplate: 0.7.0
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - nextcloud_announcements: 1.14.0
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - password_policy: 1.15.0
  - photos: 2.0.1
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.3
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - support: 1.8.0
  - survey_client: 1.13.0
  - systemtags: 1.15.0
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - updatenotification: 1.15.0
  - user_status: 1.5.0
  - video_converter: 1.0.5
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflow_media_converter: 1.7.0
  - workflowengine: 2.7.0
Disabled:
  - admin_audit
  - bruteforcesettings
  - encryption
  - suspicious_login
  - twofactor_totp
  - user_ldap
<details>
```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your server installation folder
```
</details>

**Nextcloud configuration:**
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nasdomowy.ddns.net"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "25.0.2.3",
        "overwrite.cli.url": "http:\/\/nasdomowy.ddns.net\/nextcloud",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "default_phone_region": "PL",
        "maintenance": false,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "app_install_overwrite": [
            "video_converter",
            "issuetemplate"
        ]
    }
}

**Are you using external storage, if yes which one:** local/smb/sftp/...
local
**Are you using encryption:** yes/no
no
### Server log (data/nextcloud.log)
<details>
```
[nextcloud.log](https://github.com/cwilby/nextcloud-workflow-media-converter/files/10399919/nextcloud.log)

```
</details>

**Comment by cwilby** [comment]
Thanks for the report @dominox71, I'll work on this next.

**Comment by ML-GeekHub** [comment]
Similar problem for me on Raspberry 4 NC serveur
=== Config ==============
Operating system: Ubuntu 22.04
Web server: Apache
Database: PostgreSQL
Nextcloud Version: 25.04 -- NextCloud All-In-One ( standard Docker install. )
ffmpeg 4.3.5-0
use of external storage: local
use of encryption: no
=== Scenario ==============
- Logged as admin in NextCloud
- Created a conversion test job:  conversion of 4 .avi file in mp4
- After a while, conversion batch job status switched to "converting", without CPU load increase on the serveur (rasp.).
- Two hours later, no change:  no serveur CPU load  and  displayed completed work = 0%

**Comment by ML-GeekHub** [comment]
Just in case it could help : after I cancelled the job ( = click on the trash in application window):
  - the job status widget switched to red and displayed 'failed'
  - the following message was displayed:
_(Code d'erreur 0) OCA\WorkflowMediaConverter\BackgroundJobs\BatchConvertMediaJob::findUnconvertedMediaInFolder(): Argument #1 ($folder) must be of type OCP\Files\Folder, OC\Files\Node\File given, called in /var/www/html/custom_apps/workflow_media_converter/lib/BackgroundJobs/BatchConvertMediaJob.php on line 42_

**Comment by arnolicious** [comment]
Same here, on NC 25.0.2, running on TrueNAS Scale

**Comment by cwilby** [comment]
Looking into this today (gotta love federal holidays)

**Comment by cwilby** [comment]
Hey all,

I have released v1.8.4 to Nextcloud - which includes a fix for a locking bug that was causing conversions to be re-queued / never start.

Planning to test in a wider variety of environments next, thanks for all the input!
